### PR TITLE
Code cleanup: remove 3 feature gates, deprecate 1

### PR DESCRIPTION
**Pull request by nunnatsa**
**What this PR does / why we need it**:

Removed these deprecated feature gates:
* `deployTektonTaskResources`
* `nonRoot`
* `enableManagedTenantQuota`

Deprecate the `withHostPassthroughCPU` feature gate, as it's not in use in KubeVirt, and there is no meaning by setting it in HCO.

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
None
```

**Release note**:

```release-note
Removed these deprecated feature gates:
* `deployTektonTaskResources`
* `nonRoot`
* `enableManagedTenantQuota`

Deprecate the `withHostPassthroughCPU` feature gate, as it's not in use in KubeVirt, and there is no meaning by setting it in HCO.
```

### CODE REVIEW

**Review by lyarwood — COMMENTED** [review]
/cc @xpivarc 
/cc @acardace 

Just to make you aware of the Non-Root HCO FG clean up but it looks like it's slightly overdue if anything.

File: api/v1beta1/hyperconverged_types.go [hunk 1]
```diff
@@ -446,20 +437,9 @@ type HyperConvergedFeatureGates struct {
...
-	// +optional
-	// +kubebuilder:default=true
-	// +default=true
-	NonRoot *bool `json:"nonRoot,omitempty"`
-
 	// Disable mediated devices handling on KubeVirt
 	// +optional
-	// +kubebuilder:default=false
-	// +default=false
+	// Deprecated: there is no such FG in KubeVirt. This field is ignored
```
Comment (orenc1) [inline api/v1beta1/hyperconverged_types.go:442]: i see there is such a feature gate in kubevirt:
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-config/feature-gates.go#L53

**Review by orenc1 — APPROVED** [review]
/approve

**Pull request merged**